File: OCEBook/app/src/main/java/com/devgrapher/ocebook/readium/ObjectHolder.java
package com.devgrapher.ocebook.readium;

import org.readium.sdk.android.Container;

import java.util.HashMap;
import java.util.Map;

/**
 * Created by Brent on 2/16/17.
 */

public class ObjectHolder {
    private static final ObjectHolder INSTANCE = new ObjectHolder();
    private final Map<Long, Container> containers = new HashMap<Long, Container>();
    private final Map<Long, ReadiumContext> contexts = new HashMap<Long, ReadiumContext>();

    public static ObjectHolder getInstance() {
        return INSTANCE;
    }

    public Container getContainer(Object id) {
        return containers.get(id);
    }

    public Container removeContainer(Object id) {
        return containers.remove(id);
    }

    public Container putContainer(Long key, Container value) {
        return containers.put(key, value);
    }

    public ReadiumContext getContext(Object id) {
        return contexts.get(id);
    }

    public ReadiumContext removeContext(Object id) {
        return contexts.remove(id);
    }

    public ReadiumContext putContext(Long key, ReadiumContext value) {
        return contexts.put(key, value);
    }
}

File: OCEBook/app/src/main/java/com/devgrapher/ocebook/readium/ObjectHolder.kt
package com.devgrapher.ocebook.readium

import org.readium.sdk.android.Container

import java.util.HashMap

/**
 * Created by Brent on 2/16/17.
 */

class ObjectHolder {
    private var containers = HashMap<Long, Container>()
    private var contexts = HashMap<Long, ReadiumContext>()

    fun getContainer(id: Long): Container? {
        return containers.get(id)
    }

    fun removeContainer(id: Long): Container? {
        return containers.remove(id)
    }

    fun putContainer(key: Long, value: Container): Container? {
        return containers.put(key, value)
    }

    fun getContext(id: Long): ReadiumContext? {
        return contexts.get(id)
    }

    fun removeContext(id: Long): ReadiumContext? {
        return contexts.remove(id)
    }

    fun putContext(key: Long, value: ReadiumContext): ReadiumContext? {
        return contexts.put(key, value)
    }

    companion object {
        val instance = ObjectHolder()
    }
}
